### DISCUSSION TITLE
Drawer Component

### DISCUSSION THREAD

**Discussion by kawpii** [tailwindlabs/headlessui#772] [Ideas]
Something like https://material-ui.com/components/drawers/

**Comment by croaky** [comment]
I'm interested in this as well. Another implementation is https://mantine.dev/core/drawer/ Both Mantine's `Drawer` and `Modal` components are wrapped in the concept of a `Portal` https://mantine.dev/core/portal/

**Comment by Maaverick** [comment]
Tailwind UI already provides some examples called "Slide-overs" using headlessui animation components. A good insipiration to build your own with the same tools you already know ;)

https://tailwindui.com/components/application-ui/overlays/slide-overs

**Reply by atang08** [reply]
Just not the same...trying to get this to go from left to right slide and it's taking quite a while.

**Comment by gondar00** [comment]
Any plan on adding a drawer component?

**Comment by SultanNasyrovDeveloper** [comment]
+1

**Comment by gabrieloureiro** [comment]
```tsx
const DrawerContent = forwardRef<
  ElementRef<typeof DrawerPrimitive.Content>,
  ComponentPropsWithoutRef<typeof DrawerPrimitive.Content>
>(({ className, children, ...props }, ref) => {
  const [isMounted, setIsMounted] = useState(false);

  useEffect(() => {
    if (!isMounted) {
      setIsMounted(!!document.getElementById("root")?.firstElementChild);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props]);

  return (
    <>
      {isMounted && (
        <DrawerPrimitive.Portal container={document.getElementById("theme")}>
          <DrawerOverlay />
          <DrawerPrimitive.Content
            ref={ref}
            className={cn(
              "fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col rounded-t-[10px] border bg-background",
              className,
            )}
            {...props}
          >
            <div className="mx-auto mt-4 h-2 w-[100px] rounded-full bg-foreground-secondary" />
            {children}
          </DrawerPrimitive.Content>
        </DrawerPrimitive.Portal>
      )}
    </>
  );
});
DrawerContent.displayName = "DrawerContent";
```

**Reply by gabrieloureiro** [reply]
![Screenshot 2024-05-16 at 21 04 34](https://github.com/tailwindlabs/headlessui/assets/39869298/221ad013-13dc-40ff-b7aa-7ee9408be7d9)
![Screenshot 2024-05-16 at 21 04 45](https://github.com/tailwindlabs/headlessui/assets/39869298/4fb8cf51-111f-4113-9dcf-e3de7ebb558a)

**Comment by Supcar27** [comment]
I already created a different discussion #3224 but i havent gotten any responses. I used the Dialog approach to build my own drawer and i'm stuck trying to get react-toastify to work. basically if there is a toast on the part of screen where the drawer is interaction with the toast is blocked, 

Any ideas🙏 ?
